### ISSUE TITLE
nested capture groups

### ISSUE THREAD

**Issue by charlierudolph** [cucumber/cucumber#204]
Currently having a nested capture group regexp throws an error in the javascript version (see cucumber/cucumber-js#825)
```js
// Two capture groups. The outer one should typically be non-capturing but doesn't need to be
/^a user( named "([^"]*)")?$/

// Workaround is to make the outer on non-capturing
/^a user(?: named "([^"]*)")?$/
```

I would suggest just throwing an error if there is a nested capture group and suggesting making the outer one non-capturing. Thoughts?

**Comment by charlierudolph** [comment]
The same error can be run into when defining a parameter type where the regexp contains a capture group.

**Comment by aslakhellesoy** [comment]
Yeah throwing an error with an explanation sounds like a good idea.

**Comment by aslakhellesoy** [comment]
Fixed by #227 - we're not throwing an error, just passing null instead.
